extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1088,6 +1090,9 @@ export class AmpA4A extends AMP.BaseElement {
           // modified url.
           'src': xhrFor(this.win).getCorsUrl(this.win, adUrl),
         }, SHARED_IFRAME_PROPERTIES));
+    const sentinel = generateSentinel(window);
+    const metadata = getContextMetadata(window, iframe, sentinel);
+    iframe.setAttribute('name', encodeURIComponent(JSON.stringify(metadata)));
```

Comment:
DOM manipulation doesn't work that way. `iframe.setAttribute('name', someString)` will never change anything other than the value of `iframe.name` no matter what's in `someString`.